crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -2017,6 +2018,12 @@ impl DevInspectResults {
     }
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+pub enum SuiTransactionBuilderMode {
+    Normal,
```

Comment:
I know this is the nomenclature we use within the adapter, but I wonder whether `Normal` is clear enough for what this does (i.e. `Normal` is quite a relative concept and what we currently call the normal mode is only that way because it's what existed before and `DevInspect` is new, but someone coming at this fresh might consider `DevInspect` to be the normal behaviour).

Not sure what a good name is to replace it though, here's some options:
- `ForEffects`
- `OnChain`
- `Commit`